### PR TITLE
Added Jitpack config for Java 11

### PR DESCRIPTION
**Pull request by amatsegor**

### CODE REVIEW

**Comment by amatsegor** [comment]
Same old story, but now with Jitpack

**Comment by abd3lraouf** [comment]
Don't merge this yet.

**Comment by abd3lraouf** [comment]
Working on a better CI

**Comment by abd3lraouf** [comment]
@amatsegor How can I append to your commits ?

**Comment by amatsegor** [comment]
@AbdElraoufSabri of course, feel free
Actually, @zsoltk already created a jitpack config, so I think I can close this PR and you can fork just from `master`

**Comment by zsoltk** [comment]
I'm having some issues with jitpack. Even after adding the config the build fails. https://jitpack.io/com/github/zsoltk/compose-router/b530cc0/build.log

Any idea how to fix this? @amatsegor @AbdElraoufSabri

**Comment by abd3lraouf** [comment]
@zsoltk  working on this right now.

**Comment by amatsegor** [comment]
@zsoltk 
I've added some basic maven-publish configuration and it seems to be working, build & publish process on Jitpack went fine.
https://jitpack.io/com/github/amatsegor/compose-router/0.24.1/build.log

**Comment by zsoltk** [comment]
Thanks! I'll have a look.

File: router/build.gradle [hunk 1]
```diff
@@ -58,8 +58,8 @@ afterEvaluate {
                 from components.release
 
                 group = "com.github.zsoltk"
-                artifactId = "final"
-                version = "0.24.0"
+                artifactId = "compose-router"
+                version = "0.24.1"
```
Comment (zsoltk) [inline router/build.gradle]: Does this have to be maintained to work? So far this part was automatic. I'm thinking this is potentially easy to forget about. But I guess it doesn't work without this for some reason?

Comment (amatsegor) [reply router/build.gradle]: Not sure actually, working with publication for the first time as well
@AbdElraoufSabri could you please give a hint?

Comment (amatsegor) [reply router/build.gradle]: UPD: yes, it works well without these parameters. Will remove them and push a commit

Comment (abd3lraouf) [reply router/build.gradle]: Sure, Those are for the maven central plugin I think, Not the jitpack. 

Really was thinking of providing the library through maven central. I can provide it under my namespace for example `tech.abd3lraouf:compose-router:x.y.z`.

Also was thinking of creating the library with dynamic versioning from commit messages, I've little exp with it but I maybe migrate all my open source libraries to use this approach.

Comment (zsoltk) [reply router/build.gradle]: If it does work without them then the only difference is a newline at the end of `jitpack.yml`? I need to test that out, big if true.

Comment (zsoltk) [reply router/build.gradle]: Still fails. https://jitpack.io/com/github/zsoltk/compose-router/4bc8250/build.log 

What's your secret sauce? :) Maybe the other changes are needed after all.

Comment (amatsegor) [reply router/build.gradle]: No, I meant you don't need to explicitly specify `groupId`, `artifactId` and `version`, those 3 lines solely. Applying the `maven-publish` and creating a `release(MavenPublication)` publication is still needed :D

**Review by zsoltk — APPROVED** [review]

**Comment by zsoltk** [comment]
Available as 0.24.2 from Jitpack, working now. Thank you @amatsegor @AbdElraoufSabri!

**Comment by abd3lraouf** [comment]
Yaaay Finally 👯‍♂️ Thank you all for your work @zsoltk  @amatsegor

**Pull request merged**